packages/core/data-transfer/src/strapi/providers/remote-source/index.ts

```diff
@@ -167,57 +172,29 @@ class RemoteStrapiSourceProvider implements ISourceProvider {
           }
 
           // Writes the asset's data chunks to their corresponding stream
-          else if (action === 'stream') {
-            // If the asset hasn't been registered, or if it's been closed already, then ignore the message
+          // "end" is considered a chunk, but it's not a data chunk, it's a control message
```

Comment:
The refactoring in this file is fixing a case where, when two payloads were sent at once, the first part of the first payload started writing immediately, the second payload pushed to the queue, then the second part of the first payload is pushed to the queue after.

Now instead of trying to manage locking ourself, we just -always- push to the queue in the order it comes in, and then notify the queue processor that more pieces of an asset have come in. This required switching from the idea of queuing only data to instead queueing control messages, so that we can easily ensure 'end' actions happen at the correct time, again because we do not want to have to attempt to maintain our own locking system.